src/effects/native/metronomeeffect.cpp

```diff
@@ -89,19 +89,19 @@ void MetronomeEffect::processChannel(
     }
 
     unsigned int maxFrames;
-    if (m_pSyncParameter->toBool() && groupFeatures.has_beat_length_sec) {
-        maxFrames = bufferParameters.sampleRate() * groupFeatures.beat_length_sec;
+    if (m_pSyncParameter->toBool() && groupFeatures.has_bpm) {
```

Comment:
I was thinking of setting has_bpm to false in the case of 0 bpm, which I forgot to commit in the bpmcontrol file. But actually I think we should allow 0 bpm also, and return in the case of 0 bpm before processing to avoid the divide by zero error. What do you think?

Reply:
Yes, it would be more save to check only for zero in that case. No redundant info and maximum safety.